modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/internal/upgrade/v1_4_0/ListTypeUpgradeProcess.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.list.type.internal.upgrade.v1_4_0;
+
+import com.liferay.portal.kernel.model.ListTypeConstants;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.LoggingTimer;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Christopher Kian
+ */
+public class ListTypeUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (LoggingTimer loggingTimer = new LoggingTimer();
+			PreparedStatement preparedStatement = connection.prepareStatement(
+				"select listTypeId, type_ from ListType where name IS NULL " +
+					"OR name = ''");
+			ResultSet resultSet = preparedStatement.executeQuery()) {
+
+			while (resultSet.next()) {
+				long listTypeId = resultSet.getLong("listTypeId");
+				String type = resultSet.getString("type_");
```

Comment:
Can we add the types also as condition in the sql? like 
```... where (name IS NULL or name = '') and (type_ = '" + ListTypeConstants.CONTACT_PREFIX + "' or type_ = '"+ ListTypeConstants.CONTACT_SUFFIX +"')``` ?

In this way we will loop on fewer items

Or probably is better to perform directly two updates without the select and the loop like
```update Contact_ set prefixListTypeId = 0 where prefixListTypeId in (select listTypeId from ListType where (name IS NULL or name = '') and (type_ = '"+ListTypeConstants.CONTACT_PREFIX+"'))```
and 
```update Contact_ set suffixListTypeId = 0 where suffixListTypeId in (select listTypeId from ListType where (name IS NULL or name = '') and (type_ = '"+ListTypeConstants. CONTACT_SUFFIX +"'))```

What do you think?